src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -949,8 +949,28 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
                     // we are targeting a particular package - there is no need therefore to alter other aspects of the project
                     // specifically an unrelated package may have been force removed in which case we should be happy to leave things that way
 
+                    var newPackagesToUninstallUpdated = new List<PackageIdentity>();
+                    foreach (var oldInstalledPackage in oldListOfInstalledPackages)
```

Comment:
I feel single Linq expression with many conditions is more complicated to understand and also linq degrades performance. But agree for your comments part, will add more comments